### PR TITLE
Add powerpc-unknown-linux-muslspe compile target

### PR DESCRIPTION
**Pull request by BKPepe**
This is almost identical to already existing targets:
- powerpc_unknown_linux_musl.rs
- powerpc_unknown_linux_gnuspe.rs

It has support for PowerPC SPE (muslspe), which
can be used with GCC version up to 8. It is useful for Freescale or IBM cores like e500.

This was verified to be working with OpenWrt build system for CZ.NIC's Turris 1.x routers, which are using Freescale P2020, e500v2, so add it as a Tier 3 target.

Follow-up of https://github.com/rust-lang/rust/pull/100860

### CODE REVIEW

**Comment by BKPepe** [comment]
> A tier 3 target must have a designated developer or developers (the "target maintainers") on record to be CCed when issues arise regarding the target. (The mechanism to track and CC such developers may evolve over time.)
 
@BKPepe
 
>  Targets must use naming consistent with any existing targets; for instance, a target for the same CPU or OS as an existing Rust target should use the same name for that CPU or OS. Targets should normally use the same names and naming conventions as used elsewhere in the broader ecosystem beyond Rust (such as in other toolchains), unless they have a very good reason to diverge. Changing the name of a target can be highly disruptive, especially once the target reaches a higher tier, so getting the name right is important even for a tier 3 target.
 
I have not change the name of the target, I created in similar way muslspe as it is done for gnuspe, so it is consistent with any other existing targets. 
 
 > Tier 3 targets may have unusual requirements to build or use, but must not create legal issues or impose onerous legal terms for the Rust project or for Rust developers or users.
 
Yep, that's correct. 
 
 > Neither this policy nor any decisions made regarding targets shall create any binding agreement or estoppel by any party. If any member of an approving Rust team serves as one of the maintainers of a target, or has any legal or employment requirement (explicit or implicit) that might affect their decisions regarding a target, they must recuse themselves from any approval decisions regarding the target's tier status, though they may otherwise participate in discussions.
 
 Agreed
 
> Tier 3 targets should attempt to implement as much of the standard libraries as possible and appropriate (core for most targets, alloc for targets that can support dynamic memory allocation, std for targets with an operating system or equivalent layer of system-provided functionality), but may leave some code unimplemented (either unavailable or stubbed out as appropriate), whether because the target makes it impossible to implement or challenging to implement. The authors of pull requests are not obligated to avoid calling any portions of the standard library on the basis of a tier 3 target not implementing those portions.

Support for this target should be identical to the existing powerpc support such as:
- powerpc-unknown-linux-gnu/spe
- powerpc-unknown-linux-musl
 
 > The target must provide documentation for the Rust community explaining how to build for the target, using cross-compilation if possible. If the target supports running binaries, or running tests (even if they do not pass), the documentation must explain how to run such binaries or tests for the target, using emulation if possible or dedicated hardware if necessary.
 
My use-case is to compile rust for Turris 1.x routers, which runs OpenWrt as explained in platform documentation support. 
 
> Tier 3 targets must not impose burden on the authors of pull requests, or other developers in the community, to maintain the target. In particular, do not post comments (automated or manual) on a PR that derail or suggest a block on the PR based on a tier 3 target. Do not send automated messages or notifications (via any medium, including via @) to a PR author or others involved with a PR regarding a tier 3 target, unless they have opted into such messages.
 
Agreed
 
 > Patches adding or updating tier 3 targets must not break any existing tier 2 or tier 1 target, and must not knowingly break another tier 3 target without approval of either the compiler team or the maintainers of the other tier 3 target.
 
Agreed
 
> Tier 3 targets must be able to produce assembly using at least one of rustc's supported backends from any host target.
 
 It does.

**Comment by tgross35** [comment]
Cc @wesleywiser who did an initial review in https://github.com/rust-lang/rust/pull/100860

**Comment by workingjubilee** [comment]
> tidy error: tests/assembly/targets/: missing assembly test for powerpc_unknown_linux_muslspe

**Comment by BKPepe** [comment]
Yeah, I see that. Will look into it.

**Comment by BKPepe** [comment]
It should be fixed now.

**Review by wesleywiser — APPROVED** [review]

**Comment by wesleywiser** [comment]
Thanks @BKPepe!

@bors r+ rollup

**Pull request merged**
